contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -113,86 +107,48 @@ contract FraudProofFacet is StateChannelCommon {
     function _handleBlockInvalidStateTransition(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
         BlockInvalidStateTransitionProof memory blockInvalidSTProof = abi.decode(encodedProof, (BlockInvalidStateTransitionProof));
         Block memory fraudBlock = abi.decode(blockInvalidSTProof.invalidBlock.encodedBlock, (Block));
-        DisputeAuditingData memory disputeAuditingData = blockInvalidSTProof.disputeFraudData;
-        Dispute memory dispute = blockInvalidSTProof.dispute;
+        Block memory previousBlock = abi.decode(blockInvalidSTProof.previousBlock.encodedBlock, (Block));
+        StateSnapshot memory previousBlockStateSnapshot = blockInvalidSTProof.previousBlockStateSnapshot;
+        bytes memory previousStateStateMachineState = blockInvalidSTProof.previousStateStateMachineState;
```

Comment:
You forgot to check the case blockHeight == 0. If it's not 0, then you have a previous block (and that's when you should decode it - again optional). 

since you may not have a previous block rename 'previousBlockStateSnapshot' into 'previousStateSnapshot', since in the case that height == 0 this is the genesis snapshot